src/event.rs

```diff
@@ -907,6 +908,228 @@ pub type AxisId = u32;
 /// Identifier for a specific button on some device.
 pub type ButtonId = u32;
 
+/// Cursor type.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub enum CursorType {
+    Mouse,
+    Pen(ToolState),
+    Eraser(ToolState),
+    /// Unknown cursor type. E.g. cursor was moved by a voice-recognition software or a keyboard
+    /// interaction.
+    Unknown {
+        /// Name of the cursor type.
+        name: Option<SmolStr>,
+        state: Option<ToolState>,
+    },
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub struct ToolState {
```

Comment:
I would be glad for an alternative to the term "Tool" or "State".
Originally got the idea from [Linux](https://github.com/torvalds/linux/blob/933069701c1b507825b514317d4edd5d3fd9d417/include/uapi/linux/input-event-codes.h#L402-L406), because we need a term that applies to all sorts of different tools.
But honestly I think "Pen" would have been fine as well?

Maybe "Stylus" would fit all the possible tools:
- Pen
- Eraser
- Brush
- Pencil
- Airbrush